subprojects/core/src/main/java/org/gradle/api/tasks/Copy.java

```diff
@@ -88,6 +90,23 @@ public DestinationRootCopySpec getRootSpec() {
         return (DestinationRootCopySpec) super.getRootSpec();
     }
 
+    /**
+     * Uses {@link Untracked} for {@link #getDestinationDir()}.
+     * <p>
+     * This causes the task to ignore already existing files in the destination directory.
+     * You should use this if the destination directory contains unrelated unreadable files.
+     * </p>
+     *
+     * @since 7.3
+     */
+    @Incubating
+    public void doNotTrackDestinationDir() {
```

Comment:
I think we are too much focused here on emulating an `@Untracked` annotation on `destinationDir`. Instead, what the user cares about is that Gradle shouldn't track outputs for their `Copy` task.

I think `doNotTrackOutput()` would be a better name for the method. And the Javadoc should be talking about the effects of calling the method, not the (emulated) mechanism.

Something along the lines of:

```suggestion
    /**
     * Force Gradle to ignore content in the destination directory.
     * This can be useful if the destination directory contains unreadable content, or special files like named pipes that Gradle cannot snapshot.
     * It can also be useful when copying to a location not exclusively owned by the build, for example to a common installation directory.
     * In the latter case calling this method will prevent Gradle from trying to snapshot a potentially large amount of content (that might also contain unreadable files).
     * Copy tasks with untracked outputs are never up-to-date.
     *
     * @see Untracked
     *
     * @since 7.3
     */
    @Incubating
    public void doNotTrackOutputs() {
```

Reply:
I think you are making a very good point: The user should not care about how the Copy task let's them achieve their goal, they should care about what they want to do.

I wonder if `do not track outputs` is what the user wants to achieve. It seems like the Javadoc you propose does not talk about tracking at all.

How about we use something in-line with the first line in the proposed Javadoc:
`ignoreContentInDestination`. WDYT?

Reply:
I like where this is going, though `ignoreContentInDestination` doesn't sound completely right. "Destination" implies "at", not "in", but `ignoreContentAtDestination` doesn't sound right either. I've been thinking about `ignoreOutputs`, but that is too general.

Maybe `ignoreExistingOutputs`?

Reply:
I left of the `Dir`, since I thought it was redundant. Then "in" makes sense again. So it would be `ignoreContentInDestinationDir`? I don't particularly like "outputs", since that is not very specific to the copy task. I like the `ignoreExisting`. So it could also be `ignoreExistingFilesInDestinationDir()`?

Reply:
How about `ignoreExistingContentInDestinationDir`? Though I'm not super happy about `Dir`, I think we should call things `Directory` not `Dir`, but that's another consistency issue in our APIs.

Reply:
`ignoreExistingContentInDestinationDir` it is.